### PR TITLE
@/techlabblog mdxjs

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

This PR standardise the use of mdxjs for all things markdown in @/techlabblog app from the current soupe of libraries.

 - [x] Same configuration is used in Next.js' next.config.mjs as in our own lib/data.
 - [x] [Frontmatter](https://mdxjs.com/guides/frontmatter/) and [syntax highlighting](https://mdxjs.com/guides/syntax-highlighting/) uses recommendations from mdxjs.

## Type of change

- [x] Chore

## Screenshots

Screen Shot 2024-09-30 at 09 40 00

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by kilemensi** [comment]
Happy to hear your thoughts @CodeForAfrica/tech before finalising the PR.

File: apps/techlabblog/content/publication/posts/javascript-generators.mdx [hunk 1]
```diff
@@ -11,9 +13,9 @@ Generators are a special type of function in JavaScript that can be paused and r
 
 ### Syntax
 
-A generator function is defined using the `function*`syntax, where the asterisk (\*) distinguishes it from a regular function. Inside the function body, the yield keyword is used to pause the function and return a value.
+A generator function is defined using the `function*` syntax, where the asterisk (\*) distinguishes it from a regular function. Inside the function body, the yield keyword is used to pause the function and return a value.
 
-```javascript
+```js showLineNumbers {1-3,4} caption="Caption is here" title="Title is here"
```
Comment (kelvinkipruto) [inline apps/techlabblog/content/publication/posts/javascript-generators.mdx:18]:
```suggestion
```js showLineNumbers {1-3,4} caption="Generator Example" title="Syntax"
```

File: apps/techlabblog/content/publication/authors/kelvinkipruto.mdx [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+fullName: Kelvin Kipruto
+twitter: kelvinkipruto
```
Comment (kelvinkipruto) [inline apps/techlabblog/content/publication/authors/kelvinkipruto.mdx:3]:
```suggestion
twitter: kkkipruto
```

**Review by kelvinkipruto — COMMENTED** [review]
Looks good @kilemensi, just the Typescript errors.
I see that the copy code button was removed. I think we should have it back.

File: apps/techlabblog/content/publication/posts/javascript-generators.mdx (hunk 1 above)
Comment (kilemensi) [reply apps/techlabblog/content/publication/posts/javascript-generators.mdx:18]: Yeah, this was just to show it's possible to show caption/title... This whole blogspot will be deleted anyways.

File: apps/techlabblog/content/publication/authors/kelvinkipruto.mdx (hunk 2 above)
Comment (kilemensi) [reply apps/techlabblog/content/publication/authors/kelvinkipruto.mdx:3]: Just an example Mr. KKK

**Comment by kilemensi** [comment]
Any other feedback @CodeForAfrica/tech ? Happy with this direction?

**Comment by koechkevin** [comment]
🚀

File: apps/techlabblog/content/publication/authors/kelvinkipruto.mdx (hunk 2 above)
Comment (kelvinkipruto) [reply apps/techlabblog/content/publication/authors/kelvinkipruto.mdx:3]: None. 🚀

**Pull request merged**